mlflow/crewai/autolog.py

```diff
@@ -0,0 +1,228 @@
+import inspect
+import json
+
+import mlflow
+from mlflow.entities import SpanType
+from mlflow.entities.span import LiveSpan
+from mlflow.tracing.utils import TraceJSONEncoder
+from mlflow.utils.autologging_utils.config import AutoLoggingConfig
+
+
+def patched_class_call(original, self, *args, **kwargs):
+    config = AutoLoggingConfig.init(flavor_name=mlflow.gemini.FLAVOR_NAME)
+
+    if config.log_traces:
+        fullname = f"{self.__class__.__name__}.{original.__name__}"
+        with mlflow.start_span(
+            name=fullname,
+            span_type=_get_span_type(self.__class__, fullname),
+        ) as span:
+            inputs = _construct_full_inputs(original, self, *args, **kwargs)
+            span.set_inputs(inputs)
+            _set_span_attributes(span=span, instance=self)
```

Comment:
Let's wrap this function within try-catch. In case any exception happens e.g. `AttributeError`, we don't want to drop the span entirely but instead gracefully skip that attributes.